lint-consul-retry.go

```diff
@@ -93,42 +90,49 @@ type visitor struct {
 	depth       int
 	currentTest string
 	path        string
+	retryDepth  int
+	requireNew  bool
 }
 
 func (v visitor) Visit(n ast.Node) ast.Visitor {
-	// Walk uses DFS so reset when we pop back up
-	if retryDepth > 0 && v.depth <= retryDepth {
-		retryDepth = 0
+	if n != nil {
+		// When called with a non-nil ast.Node, we are delving 1 node deeper into the tree
+		// and therefore should update our tracked depth accordingly.
+		v.depth++
+	} else {
+		// Once a sub-tree of the AST has finished being walked Visit(nil) will be invoked.
+		// There is no need to decrement the depth because we are passing the visitor by
+		// value rather than a reference. The previous visitor on up the stack with the correct
+		// depth would be used for subsequent recursive Visit calls.
+		return v
 	}
 
 	switch node := n.(type) {
 	case *ast.CallExpr:
 		if inRequire(node) {
-			newRequire = true
+			v.requireNew = true
 		}
-		if inRetry(node) {
-			retryDepth = v.depth
+
+		isRetryInvocation := inRetry(node)
+		if isRetryInvocation {
+			v.retryDepth = v.depth
 		}
-		if retryDepth > 0 && tCallsFailer(node.Fun) {
+
+		if v.retryDepth > 0 && tCallsFailer(node.Fun) {
```

Comment:
It was still needed due to the differences in how the AST represents a method invocation vs a function call and where the `t` will show up. I have refactored the code so that usesTestingT will find it for both cases.